src/lib.rs

```diff
@@ -1684,6 +1684,84 @@ pub mod gl {
                 },
             }
         }
+
+        pub fn gen_samplers(&self, n: GLsizei) -> Vec<GLuint> {
+            let mut result = vec![0 as GLuint; n as usize];
+            match self {
+                Gl::Gl(gl) => unsafe { gl.GenSamplers(n, result.as_mut_ptr()) },
+                Gl::Gles(gles) => unsafe { gles.GenSamplers(n, result.as_mut_ptr()) },
+            };
+            result
+        }
+
+        pub fn delete_samplers(&self, samplers: &[GLuint]) {
+            match self {
+                Gl::Gl(gl) => unsafe { gl.DeleteSamplers(samplers.len() as GLsizei, samplers.as_ptr()) },
+                Gl::Gles(gles) => unsafe { gles.DeleteSamplers(samplers.len() as GLsizei, samplers.as_ptr()) },
+            }
+        }
+
+        pub fn is_sampler(&self, sampler: GLuint) -> GLboolean {
+            match self {
+                Gl::Gl(gl) => unsafe { gl.IsSampler(sampler) },
+                Gl::Gles(gles) => unsafe { gles.IsSampler(sampler) },
+            }
+        }
+
+        pub fn bind_sampler(&self, target: GLenum, sampler: GLuint) {
+            match self {
+                Gl::Gl(gl) => unsafe { gl.BindSampler(target, sampler) },
+                Gl::Gles(gles) => unsafe { gles.BindSampler(target, sampler) },
+            }
+        }
+
+        pub fn get_sampler_parameter_iv(&self, sampler: GLuint, pname: GLenum) -> Vec<GLint> {
+            let mut result = vec![0 as GLint];
```

Comment:
What about GL_TEXTURE_BORDER_COLOR? Is there a risk it could be used, which would exceed the bounds of this array?

Reply:
Nevermind, I read the wrong documentation page.